### ISSUE TITLE
[BUG] Node type (container, vm, bare-metal) logic is incorrect

### ISSUE THREAD

**Issue by dionysis-netdata** [netdata/netdata-cloud#305]
There are cases where our calculation for a node type is incorrect. Example is latest ticket raised by @thiagoftsm #299 where nodes are marked as `container`. Same as the examples raised by @ktsaou under #300.

The examples on those tickets should have been bare-metal. They have a value of `container: unknown` and as per our implementation logic they are marked as containers (incorrectly). 

I also noticed that our logic also doesn't make use of `container_detection` values.

<img width="383" alt="image" src="https://user-images.githubusercontent.com/95340881/155357622-96ac14d9-5848-4847-a1e7-f8a5f6c2fd57.png">

### Current Implementation 
Here are values exported from all agents by count and how our logic works right now:

It is sequential and starts by checking if the node is a container from the `container` attribute

container | count | Type
-- | -- | --
docker | 286.2K | docker
none | 245.7K |  
unknown | 27.9K | container
other | 16.7K | container
container | 4.9K | container
lxc | 2.9K | lxc
  | 583 |  
openvz | 569 | openvz
container-other | 136 | container
wsl | 114 | wsl
podman | 41 | podman
systemd-nspawn | 15 | systemd-nspawn
lxc-libvirt | 3 | lxc-libvirt
null | 1 |  

If no container is found it checks for Virtualisation

Virtualisation | Count | Type
-- | -- | --
none | 235.7K | baremetal
kvm | 167K | kvm
hypervisor | 135K | hypervisor
vmware | 19.7K | vmware
xen | 9.7K | xen
microsoft | 9.5K | microsoft
amazon | 5.2K | amazon
oracle | 1.6K | oracle
qemu | 1.3K | qemu
  | 402 | baremetal
unknown | 181 | vm
bhyve | 101 | bhyve
$(systemd-detect-virt -v) | 32 | vm
other | 25 | vm
bochs | 23 | bochs
vm-other | 22 | vm
none #vmware | 11 | vmware
$(systemd-detect-virt -v) docker-compose up | 6 | vm
lxc | 6 | vm
detect-virt -v | 4 | vm
systemd-detect-virt -v | 3 | vm
parallels | 2 | parallels
wmware | 2 | vmware
VMware Virtual Platform | 1 | vmware
zvm | 1 | zvm
(systemd-detect-virt -v) | 1 | vm
${VIRTUALIZATION} | 1 | vm
null | 1 | baremetal

If no match is found type is considered `baremetal`

@thanos-karachalios can we also run the following query on nodes ?
```
SELECT system_virtualization, system_virtualization_detection, system_container, system_container_detection, count(*)
FROM NODES
GROUP BY system_virtualization, system_virtualization_detection, system_container, system_container_detection
```

Notes
* https://github.com/netdata/netdata-cloud/issues/300
* https://github.com/netdata/netdata-cloud/issues/299
* https://github.com/netdata/cloud-frontend/issues/2889
* https://github.com/netdata/cloud-frontend/issues/2903
* https://docs.google.com/spreadsheets/d/1QlCEah1eaksyOkpOKCpc5yvmxpFiGvGL7aw1C63LLzA/edit#gid=0

**Comment by dionysis-netdata** [comment]
@cpipilas , @stelfrag Can you verify or comment on our calculation algorithm ?

**Comment by ktsaou** [comment]
@dionysis-netdata I suggest to go at #299 and #300 and ask your team to post there the json objects of the nodes, as they have them. Once they post these data, we will find the bug.

PS: This is another reason, why the json objects should be available at the UI. Now we have to ask for them...

**Comment by dionysis-netdata** [comment]
To recap: The calculation implemented on Cloud FE under https://github.com/netdata/cloud-frontend/issues/2889
treats `container: unknown` in Cloud as type `container`.

This is the case for both cases reported under #299 and #300.

As per @ktsaou 's comment https://github.com/netdata/netdata-cloud/issues/300#issuecomment-1049150109
a value of `container: unknown` shouldn't NOT be of type `container`

**Comment by burbuli8ra** [comment]
@dionysis-netdata and @ktsaou since I am doing the update, I noticed that we treat the keyword `unknown` as `vm` in `virtalization` as well?
Do we maybe want to treat `virtualization: unknown` as well as no virtual machine?
Please note that this means that the label will be `Bare metal` in that case!

**Comment by ktsaou** [comment]
@burbuli8ra leave it as is. Let's find a case that is wrong.
I see above, we have detected only 181 nodes with `unknown` virtualization so far, unlike the 28k with `unknown` containers.

**Comment by burbuli8ra** [comment]
Thank you @ktsaou !

**Comment by thanos-karachalios** [comment]
@dionysis-netdata 

here are some data for the request. Tables are from `vw_node360` and `node_nodes_info_latest` (not limiting to reachable nodes or not as discussed)



Row | virtualization_type | container_type | counts
-- | -- | -- | --
1 | null | null | 19 |  
2 | $(systemd-detect-virt -v) | docker | 1 |  
3 | $(systemd-detect-virt -v) docker-compose up | docker | 1 |  
4 | amazon | none | 903 |  
5 | amazon | unknown | 2 |  
6 | bhyve | docker | 1 |  
7 | bhyve | none | 25 |  
8 | bhyve | unknown | 3 |  
9 | bochs | none | 15 |  
10 | bochs | unknown | 8 |  
11 | bochs | docker | 1 |  
12 | hypervisor | docker | 9687 |  
13 | hypervisor | unknown | 15356 |  
14 | hypervisor | container | 312 |  
15 | hypervisor | lxc | 6 |  
16 | kvm | none | 21128 |  
17 | kvm | unknown | 1739 |  
18 | kvm | docker | 144 |  
19 | kvm | lxc | 28 |  
20 | kvm | openvz | 17 |  
21 | kvm | wsl | 1 |  
22 | kvm | systemd-nspawn | 1 |  
23 | microsoft | none | 1246 |  
24 | microsoft | wsl | 54 |  
25 | microsoft | unknown | 108 |  
26 | microsoft | docker | 11 |  
27 | none | docker | 6100 |  
28 | none | none | 16547 |  
29 | none | unknown | 2115 |  
30 | none | lxc | 1002 |  
31 | none | container | 71 |  
32 | none | openvz | 255 |  
33 | none | wsl | 42 |  
34 | none | null | 10 |  
35 | none | systemd-nspawn | 2 |  
36 | none | container-other | 1 |  
37 | oracle | none | 302 |  
38 | oracle | unknown | 12 |  
39 | oracle | lxc | 1 |  
40 | other | none | 2 |  
41 | parallels | none | 5 |  
42 | qemu | none | 1067 |  
43 | qemu | unknown | 66 |  
44 | qemu | docker | 1 |  
45 | unknown | docker | 120 |  
46 | unknown | unknown | 24 |  
47 | unknown | null | 26 |  
48 | unknown | none | 14 |  
49 | vm-other | none | 31 |  
50 | vm-other | unknown | 1 |  
51 | vmware | none | 4198 |  
52 | vmware | unknown | 552 |  
53 | vmware | docker | 19 |  
54 | vmware | lxc | 3 |  
55 | vmware | null | 20 |  
56 | wmware | docker | 20 |  
57 | xen | none | 2556 |  
58 | xen | unknown | 193 |  
59 | xen | docker | 7 |  
60 | zvm | none | 1

**Comment by burbuli8ra** [comment]
Released on [v2.17.2](https://github.com/netdata/cloud-frontend/releases/tag/2.17.2)

**Comment by burbuli8ra** [comment]
Hello @ktsaou , the fix for `virtualization = unknown` has been released on [v2.18.2](https://github.com/netdata/cloud-frontend/releases/tag/2.18.2)

**Comment by ktsaou** [comment]
Great! Thank you @burbuli8ra !
